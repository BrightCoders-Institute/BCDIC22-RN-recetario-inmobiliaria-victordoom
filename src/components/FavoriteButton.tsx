import React from 'react'
import { StyleSheet, View } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

export const FavoriteButton = () => {
    return (
        <View style={styles.container}>
            <Icon name='heart' size={20} color='#fff' />
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        height: 30,
        width: 30,
        backgroundColor: 'green',
        borderRadius: 20,
        justifyContent: 'center',
        alignItems: 'center'
    }
});
