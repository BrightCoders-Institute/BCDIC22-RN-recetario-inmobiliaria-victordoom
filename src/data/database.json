[
	{
        "address": "100 Avenida Prospect a 6400 Hollywood",
        "bathrooms": 1,
        "cost": 417,
        "favorite": false,
        "image": "https://ap.rdcpix.com/6772053682b695a4143ad6be12b3e050l-m2747259734od-w480_h360_x2_s1.jpg",
        "name": "The Hollywood",
        "qualification": 4,
        "rooms": 5,
        "surface": {
           "lenght": 21,
           "width": 31
        }
     },
     {
        "address": "100 Avenida Prospect a 6400 Hollywood",
        "bathrooms": 3,
        "cost": 439,
        "favorite": true,
        "image": "https://ap.rdcpix.com/6772053682b695a4143ad6be12b3e050l-m2747259734od-w480_h360_x2_s1.jpg",
        "name": "The Hollywood",
        "qualification": 4,
        "rooms": 6,
        "surface": {
           "lenght": 12,
           "width": 27
        }
     },
     {
        "address": "100 Avenida Prospect a 6400 Hollywood",
        "bathrooms": 3,
        "cost": 478,
        "favorite": false,
        "image": "https://ap.rdcpix.com/6772053682b695a4143ad6be12b3e050l-m2747259734od-w480_h360_x2_s1.jpg",
        "name": "The Hollywood",
        "qualification": 1,
        "rooms": 2,
        "surface": {
           "lenght": 19,
           "width": 33
        }
     },
     {
        "address": "100 Avenida Prospect a 6400 Hollywood",
        "bathrooms": 3,
        "cost": 479,
        "favorite": true,
        "image": "https://ap.rdcpix.com/6772053682b695a4143ad6be12b3e050l-m2747259734od-w480_h360_x2_s1.jpg",
        "name": "The Hollywood",
        "qualification": 4,
        "rooms": 6,
        "surface": {
           "lenght": 18,
           "width": 20
        }
     },
     {
        "address": "100 Avenida Prospect a 6400 Hollywood",
        "bathrooms": 1,
        "cost": 451,
        "favorite": true,
        "image": "https://ap.rdcpix.com/6772053682b695a4143ad6be12b3e050l-m2747259734od-w480_h360_x2_s1.jpg",
        "name": "The Hollywood",
        "qualification": 4,
        "rooms": 5,
        "surface": {
           "lenght": 21,
           "width": 30
        }
     }
]