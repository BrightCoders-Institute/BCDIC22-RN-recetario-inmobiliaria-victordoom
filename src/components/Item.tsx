import React from 'react'
import { Image, StyleSheet, Text, View } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
import { RentalProperties } from '../interface/rentalPropertiesInterface'
import { FavoriteButton } from './FavoriteButton'


interface Props {
  rental: RentalProperties
}
export const Item = ({ rental }: Props) => {

  const { image, address, name, bathrooms, rooms, cost, surface, qualification } = rental;
  return (
    <View style={styles.contanier}>
      <View style={styles.imageContainer}>
        <Image
          source={{ uri: image }}
          style={styles.imageProp}
        />
        <View style={styles.starBadge}>
          <Icon name='star' size={20} color='#EEBA00' />
          <Text style={styles.numberstart}>{qualification}</Text>
        </View>
      </View>
      <View style={{ flex: 1 }}>
        <Text style={styles.title}>{name}</Text>
        <View style={styles.rowIcons}>
          <Icon name='map-marker-outline' size={20} color='#B4B8B9' />
          <Text style={styles.label}>{address}</Text>
        </View>
        <View style={styles.rowIcons2}>
          <Icon name='bed-king-outline' size={30} color='#B4B8B9' />
          <Text style={styles.number}>{rooms}</Text>
          <Icon name='shower' size={30} color='#B4B8B9' />
          <Text style={styles.number}>{bathrooms}</Text>
          <Icon name='arrow-expand-all' size={30} color='#B4B8B9' />
          <Text style={styles.number}>{(surface.lenght * surface.width) + ' ft'}&#178;</Text>
        </View>
        <Text style={styles.title}>&#36;{cost}/m</Text>
        <View style={styles.buttonFav}>
          <FavoriteButton />
        </View>

      </View>
    </View>

  )
}

const styles = StyleSheet.create({
  contanier: {
    flexDirection: 'row',
    backgroundColor: '#f5fdff',
    margin: 10,
    borderRadius: 10,
    padding: 10
  },
  imageProp: {
    height: 95,
    width: 90,
    borderRadius: 10
  },
  rowIcons: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center'
  },
  rowIcons2: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  rowPrice: {
    flexDirection: 'row'
  },
  title: {
    fontSize: 18,
    fontWeight: 'bold'
  },
  label: {
    fontSize: 13,
    color: '#B4B8B9'
  },
  number: {
    fontSize: 13,
    marginRight: 9,
    marginLeft: 3,
    fontWeight: 'bold'
  },
  numberstart: {
    fontSize: 13,
    fontWeight: 'bold'
  },
  buttonFav: {
    position: 'absolute',
    bottom: 0,
    right: 0,
  },
  starBadge: {
    flexDirection: 'row',
    backgroundColor: '#FBEDB7',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 10,
    position: 'absolute',
    bottom: 5,
    paddingHorizontal: 10
  },
  imageContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: 10,
    marginVertical: 5
  }
});
