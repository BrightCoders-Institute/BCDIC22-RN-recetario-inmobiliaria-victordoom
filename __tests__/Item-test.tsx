import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { Item } from '../src/components/Item';
import database from '../src/data/database.json'

it('renders correctly', () => {
    renderer.create(<Item rental={database[1]} />);
});

describe('<Item/>', () => {
    test('Item snapShot', () => {
        const snap = renderer.create(
            <Item rental={database[1]} />
        ).toJSON();
        expect(snap).toMatchSnapshot();
    });
})
it('can render all the properties from the database', () => {
    jest.useFakeTimers();
    const components = [];
    for (const item of database) {
        const component = renderer.create(<Item rental={item} />).toJSON();
        if (component) {
            components.push(component);
        }
    }
    expect(components.length).toBe(database.length);
});