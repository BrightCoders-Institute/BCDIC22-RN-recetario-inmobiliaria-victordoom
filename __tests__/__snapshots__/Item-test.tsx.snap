// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> Item snapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f5fdff",
      "borderRadius": 10,
      "flexDirection": "row",
      "margin": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginRight": 10,
        "marginVertical": 5,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://ap.rdcpix.com/6772053682b695a4143ad6be12b3e050l-m2747259734od-w480_h360_x2_s1.jpg",
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "height": 95,
          "width": 90,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FBEDB7",
          "borderRadius": 10,
          "bottom": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingHorizontal": 10,
          "position": "absolute",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#EEBA00",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰓎
      </Text>
      <Text
        style={
          Object {
            "fontSize": 13,
            "fontWeight": "bold",
          }
        }
      >
        4
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      The Hollywood
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#B4B8B9",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰟙
      </Text>
      <Text
        style={
          Object {
            "color": "#B4B8B9",
            "fontSize": 13,
          }
        }
      >
        100 Avenida P. 6400 Hollywood
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#B4B8B9",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰿑
      </Text>
      <Text
        style={
          Object {
            "fontSize": 13,
            "fontWeight": "bold",
            "marginLeft": 3,
            "marginRight": 9,
          }
        }
      >
        6
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#B4B8B9",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰦠
      </Text>
      <Text
        style={
          Object {
            "fontSize": 13,
            "fontWeight": "bold",
            "marginLeft": 3,
            "marginRight": 9,
          }
        }
      >
        3
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#B4B8B9",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰁌
      </Text>
      <Text
        style={
          Object {
            "fontSize": 13,
            "fontWeight": "bold",
            "marginLeft": 3,
            "marginRight": 9,
          }
        }
      >
        324 ft
        ²
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      $
      439
      /m
    </Text>
    <View
      style={
        Object {
          "bottom": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00B074",
            "borderRadius": 20,
            "height": 30,
            "justifyContent": "center",
            "width": 30,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
